### PR TITLE
feat: create product library

### LINKED ISSUE #7983: Create product lib
As seen in #7959, we will start to introduce more smaller feature-libraries to Spartacus codebase, and we will start with `product`.

At the same time, we should create an entry point for `product-configuration`.

### PR DESCRIPTION
**Pull request by znikola**
closes #7983

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
   "version": 1,
-  "newProjectRoot": "projects",
+  "newProjectRoot": "feature-libs",
```
Comment (marlass) [inline angular.json:4]: This is to inform `ng cli` where to create new libs?

File: feature-libs/product/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "spartacus",
+    "framework",
+    "storefront",
+    "product"
+  ],
+  "license": "Apache-2.0",
+  "publishConfig": {
+    "access": "public"
+  },
+  "repository": "https://github.com/SAP/spartacus",
```
Comment (marlass) [inline feature-libs/product/package.json]: Should we point to the specific directory, like we do in core libs?
Missed that for my-account as well

File: package.json [hunk 3]
```diff
@@ -89,6 +89,7 @@
     "release:cds:with-changelog": "cd projects/cds && release-it && cd ../..",
     "build:myaccount:lib": "ng build my-account --prod",
     "release:myaccount:with-changelog": "cd feature-libs/my-account && release-it && cd ../..",
+    "build:product:lib": "ng build product --prod",
```
Comment (marlass) [inline package.json:92]: We should also add the release script, even that we won't use it in the 2.1

File: scripts/changelog.ts [hunk 4]
```diff
@@ -77,6 +77,7 @@ export default async function run(
     '@spartacus/incubator': 'projects/incubator',
     '@spartacus/cds': 'projects/cds',
     '@spartacus/my-account': 'feature-libs/my-account',
+    '@spartacus/product': 'feature-libs/product',
```
Comment (marlass) [inline scripts/changelog.ts:80]: We also need to add the lib at the enf od this file, when we detect the library by name.
Missed that for my-account as well.

**Review by marlass — CHANGES_REQUESTED** [review]

File: angular.json (hunk 1 above)
Comment (znikola) [reply angular.json:4]: Yes, correct. As we are moving away from `projects`, I think `feature-libs` should be the new project roon

**Review by marlass — APPROVED** [review]

**Pull request merged**